test/e2e/selectors/selectors_test.go

```diff
@@ -65,64 +63,82 @@ var _ = Describe("Field Selectors in Flight Control", Ordered, func() {
 	Context("Basic Functionality Tests", func() {
 		It("We can list devices and create resources", Label("77917"), func() {
 			By("Listing devices")
-			out, err := harness.CLI("get", "devices")
+			out, err := harness.RunGetDevices()
 			Expect(err).ToNot(HaveOccurred())
 			Expect(out).To(ContainSubstring("NAME"))
 
 			By("create a complete fleet")
-			_, _ = harness.CLI("delete", "fleet")
-			out, err = harness.CLI("apply", "-f", util.GetTestExamplesYamlPath(fleetYAMLPath))
+			// Creating fleet
+			_, _ = harness.ManageResource("delete", "fleet")
+			out, err = harness.ManageResource("apply", fleetYAMLPath)
+			// Verifying fleet is valid
+			Expect(err).ToNot(HaveOccurred())
+			Expect(out).To(MatchRegexp(resourceCreated))
 			Expect(err).ToNot(HaveOccurred())
 			Expect(out).To(MatchRegexp(resourceCreated))
 
 			By("Get device info from the yaml")
-			deviceInfo = harness.GetDeviceByYaml(util.GetTestExamplesYamlPath(deviceYAMLPath))
+			deviceInfo = harness.GetDeviceByYaml(GetTestExamplesYamlPath(deviceYAMLPath))
 			deviceName = *deviceInfo.Metadata.Name
+			deviceBInfo = harness.GetDeviceByYaml(GetTestExamplesYamlPath(deviceBYAMLPath))
+			deviceBName = *deviceBInfo.Metadata.Name
 			Expect(deviceName).ToNot(BeEmpty())
+			Expect(deviceBName).ToNot(BeEmpty())
 
+			// deleting previous devices
 			_, _ = harness.CLI("delete", "device")
-			out, err = harness.CLI("apply", "-R", "-f", util.GetTestExamplesYamlPath(deviceYAMLPath))
-			time.Sleep(30 * time.Second) // to establish fleet before adding device to it
-			Eventually(func() error {
-				_, err := harness.CLI("get", "fleet")
-				return err
-			}, "30s", "1s").Should(BeNil(), "Timeout waiting for fleet to be ready")
+			// Creating new devices
+			out, err = harness.ManageResource("apply", deviceYAMLPath)
 			Expect(err).ToNot(HaveOccurred())
 			Expect(out).To(MatchRegexp(resourceCreated))
-			Expect(out).To(ContainSubstring(fmt.Sprintf("%s/%s", deviceYAMLPath, deviceName)))
+			out, err = harness.ManageResource("apply", deviceBYAMLPath)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(out).To(MatchRegexp(resourceCreated))
+
+			logrus.Infof("Device Created: %s", deviceName)
+			logrus.Infof("Device Created: %s", deviceBName)
+
+			// to establish fleet before adding device to it
+			time.Sleep(30 * time.Second)
+			Eventually(func() error {
+				out, err = harness.CLI("get", "device")
+				return err
+			}, DefaultTimeout, RetryInterval).Should(BeNil(), "Timeout waiting for fleet to be ready")
+			Expect(out).To(ContainSubstring("defaultDevice"))
 
 			By("create a complete repo")
-			out, err = harness.CLI("apply", "-f", util.GetTestExamplesYamlPath(repoYAMLPath))
```

Comment:
What is the purpose of creating _repo_ here? I don't see it is used or mentioned in the TC